user/views.py

```diff
@@ -54,33 +41,37 @@ def update_user(user, request):
         else:
             password = data.get('cur_password', '')
             if not user.check_password(password):
-                raise SluglineAPIException({'user': ['USER.PASSWORD.CURRENT_INCORRECT']})
+                raise APIException({'user': ['USER.PASSWORD.CURRENT_INCORRECT']})
             del data['cur_password']
 
     # We set the partial flag as the front-end may not choose to update all fields at once
     serializer = UserSerializer(data=data, instance=user, partial=True)
     serializer.is_valid()
     if len(serializer.errors):
-        raise SluglineAPIException(serializer.errors)
+        raise APIException(serializer.errors)
     else:
         try:
             updated_user = serializer.save()
             if 'password' in data:
                 update_session_auth_hash(request, updated_user)
-            return Response({
-                'success': True,
-                'user': serializer.data
-            })
+            return Response(serializer.data)
         except Exception:
-            raise SluglineAPIException('USER.COULD_NOT_UPDATE')
+            raise APIException('USER.COULD_NOT_UPDATE')
 
 
-@api_view(['PATCH'])
-@permission_classes([IsAuthenticated])
-def update_user_view(request):
-    if not request.user.is_staff and not request.user.is_editor and any(['is_editor' in request.data]):
-        raise SluglineAPIException('USER.INSUFFICIENT_PRIVILEGES')
-    return update_user(user=request.user, request=request)
+@api_view(['GET', 'PATCH'])
+def single_user_view(request):
```

Comment:
Should call this `current_user_view` or something similar.

Reply:
09263381